server/routes/experiments/biomed_nl/traversal.py

```diff
@@ -686,21 +693,32 @@ def find_start_and_traversal_type(self):
       str_to_raw[entity.sanitized_str] = entity.raw_str
       for synonym in entity.synonyms:
         str_to_raw[synonym] = entity.raw_str
-    entities_to_dcids, entities_to_recognized_types = recognize_entities_from_query(
-        ' '.join((str_to_raw.keys())))
+
+    detected_entities = recognize_entities_from_query(' '.join(
+        (str_to_raw.keys())))
+
+    displayed_entities = {}
 
     start_dcids = set()
-    # Add dcids of only the start entity to start_dcids.
-    for entity, dcids in entities_to_dcids.items():
-      if entity in start_strs:
-        start_dcids.update(dcids)
     raw_to_types = {}
-    for entity, types in entities_to_recognized_types.items():
-      raw_to_types.setdefault(str_to_raw[entity], []).extend(types)
+    for entity in detected_entities:
+      raw_to_types.setdefault(str_to_raw.get(entity.name, entity.name),
+                              []).extend(entity.types)
+      if any(entity.name in start_str for start_str in start_strs):
+        start_dcids.add(entity.dcid)
+
+      if entity.dcid in displayed_entities:
+        combined_types = list(
+            set(displayed_entities[entity.dcid].types) | set(entity.types))
+        displayed_entities[entity.dcid].types = combined_types
```

Comment:
could there be multiple detected entities with the same dcid but different name? wouldn't this overwrite the entity's name field if there could be different names and is that ok?

Reply:
Yes! It's basically whichever name it sees first is the name we'll show in the response. It's quicker for now to just show one name for an entity instead of listing all of them, especially since we're also displaying the dcid and types, I think the chips would get a little cramped. 
We can revisit this if it becomes an issue and work out a good way to display multiple names for a single entity.